waku/v2/node/wakunode2.nim

```diff
@@ -122,14 +123,56 @@ proc removeContentFilters(filters: var Filters, contentFilters: seq[ContentFilte
 template tcpEndPoint(address, port): auto =
   MultiAddress.init(address, tcpProtocol, port)
 
+
+template addWsFlag() =
+  MultiAddress.init("ws").tryGet()
+
 ## Public API
 ##
+proc newWakuSwitch*(
+    privKey = none(PrivateKey),
+    address = MultiAddress.init("/ip4/127.0.0.1/tcp/0").tryGet(),
+    secureManagers: openarray[SecureProtocol] = [
+        SecureProtocol.Noise,
+      ],
+    transportFlags: set[ServerFlags] = {},
+    rng = crypto.newRng(),
+    inTimeout: Duration = 5.minutes,
+    outTimeout: Duration = 5.minutes,
+    maxConnections = MaxConnections,
+    maxIn = -1,
+    maxOut = -1,
+    maxConnsPerPeer = MaxConnectionsPerPeer,
+    transport: string): Switch
+    {.raises: [Defect, LPError].} =
+
+    var b = SwitchBuilder
+      .new()
+      .withAddress(address)
+      .withRng(rng)
+      .withMaxConnections(maxConnections)
+      .withMaxIn(maxIn)
+      .withMaxOut(maxOut)
+      .withMaxConnsPerPeer(maxConnsPerPeer)
+      .withMplex(inTimeout, outTimeout)
+      .withNoise()
+
+    if transport == "ws" or transport == "wss":
```

Comment:
The addition of `wss` is not consistent across the PR. I'd suggest to leave it totally out for now and add it once you do a PR to add `wss`.